### ISSUE TITLE
Request: add Casambi switch support

### ISSUE THREAD

**Issue by Anthrow** [awahlig/homebridge-casambi#17]
I have added a non casambi occhio luminaire to my casambi network by using an AImotion plug & play switch. It works perfectly in the Casambi app and the Casambi plugin for homebridge detects it perfectly fine. The switch als gets added to my HomeKit setup however pressing the switch in HomeKit does not turn the switch on or off. Checking the Homebridge logs results in the following error:

“Luminaire switch plug & play has no on characteristics”

After reading your list of tested devices I noticed that you explicitly state that on/off functionality is not supported. So it all seems to add up. Are you planning on adding this functionality or are there certain software limitation that would make this impossible to implement? It would be great to be able to successfully use this switch in my HomeKit network.

Thanks in advance,
Martijn

**Comment by awahlig** [comment]
The missing On characteristic would indeed result in this problem.

Could you send me the log when Homebridge starts up? In particular I’m looking for a line saying:

“Unit switch plug & play is a … with fixtureId …”

With that I should be able to debug the missing characteristic.

**Comment by Anthrow** [comment]
Here is a screenshot of the log: 
![5CAC7AC3-7FDE-4D32-BA52-A6AF4A4AA07A](https://user-images.githubusercontent.com/20062861/132997649-326392bd-8ec2-4ac9-ac78-584ac08260ea.jpeg)

**Comment by Anthrow** [comment]
> “Unit switch plug & play is a … with fixtureId …”

So I guess that would be 7757 :)

**Comment by awahlig** [comment]
> So I guess that would be 7757 :)

Thanks :)

I’ll get back to you when I know more.

**Comment by Anthrow** [comment]
Awesome! Many thanks in advance Awahlig!

**Comment by Anthrow** [comment]
Hello awahlig, I was wondering wether you have found a solution for the switch? Today is the last day I can return the unit and get a refund.

**Comment by awahlig** [comment]
Yes, sorry. I know why it doesn’t work and how to make it work. I haven’t had time yet to start implementing it but I will so you can keep it.

**Comment by awahlig** [comment]
I just published a new beta version to npm with support for simple on/off switches.

To try it out, you'll have to update your homebridge-casambi package:
> sudo npm install -g homebridge-casambi@beta

Then, restart Homebridge and check the Plugins page of the Homebridge Config UI to see if it's using the correct version of the plugin (v0.3.4-beta.0).

**Comment by Anthrow** [comment]
Hello there,

I returned the unit because it was unsupported and I didn’t want to risk purchasing a switch that was unusable. I will try to get my hands on one soon so i can give your beta a try.

Met vriendelijke groet,

Martijn de Groot

**Comment by awahlig** [comment]
Hey Martijn, no problem, I understand. Sorry it took this long, I was quite busy lately. If you manage to get one again, let me know.

Best,
Arkadiusz

**Comment by Anthrow** [comment]
Will do! Thanks for investing time in making this possible.

Met vriendelijke groet,

Martijn de Groot
